src/server/middleware/csrf-token.js

```diff
@@ -0,0 +1,67 @@
+import uuid from 'node-uuid';
+
+export const generateToken = (req, res, next) => {
+  // Don't generate new tokens for API calls
+  if (req.path.match(/^\/api/)) {
+    return next();
+  }
+
+  // TODO: Remove debug
+  const csrfToken = uuid.v4();
+  if (typeof req.session.csrfTokens === 'undefined') {
+    req.session.csrfTokens = [];
+  }
+
+  req.session.csrfTokens.push(csrfToken);
+
+  // Only 5 valid CSRF tokens at a time. Safe to increase number.
+  while (req.session.csrfTokens.length > 5) {
+    req.session.csrfTokens.shift();
+  }
```

Comment:
We keep the last 5 generated CSRF tokens for each user. A new token is generated on every page view that renders the application. This should be enough for the user to use the site in more than one browser tab.

Interested in some debate about whether this is five tokens is enough to keep.